Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -100,26 +100,51 @@ class TIFFLoadingContext {
         }
     }
 
+    Optional<u8> alpha_channel_index() const
+    {
+        if (m_metadata.extra_samples().has_value()) {
+            auto const extra_samples = m_metadata.extra_samples().value();
+            for (u8 i = 0; i < extra_samples.size(); ++i) {
+                if (extra_samples[i] == ExtraSample::UnassociatedAlpha)
+                    return i + samples_for_photometric_interpretation();
+            }
+        }
+        return OptionalNone {};
+    }
+
     ErrorOr<Color> read_color(BigEndianInputBitStream& stream)
     {
         auto bits_per_sample = *m_metadata.bits_per_sample();
 
         // Section 7: Additional Baseline TIFF Requirements
         // Some TIFF files may have more components per pixel than you think. A Baseline TIFF reader must skip over
         // them gracefully, using the values of the SamplesPerPixel and BitsPerSample fields.
-        auto discard_unknown_channels = [&]() -> ErrorOr<void> {
-            auto const unknown_channels = *m_metadata.samples_per_pixel() - samples_for_photometric_interpretation();
-            for (u8 i = bits_per_sample.size() - unknown_channels; i < bits_per_sample.size(); ++i)
-                TRY(read_component(stream, bits_per_sample[i]));
-            return {};
+        auto manage_extra_channels = [&]() -> ErrorOr<u8> {
+            // Both unknown and alpha channels are considered as extra channels, so let's iterate over
+            // them, conserve (if any) the alpha value and discard anything else.
```

Comment:
nit: i'd put "(if any)" after "the alpha value". And, not a native speaker, but s/anything/everything/ is more correct / sounds better here I think. (Not sure though.)